spec/frameSpec.js

```diff
@@ -0,0 +1,145 @@
+describe('Frame', () => {
+  describe('constructor', () => {
+    it('following values are set upon initialisation', () => {
+      let frame = new Frame;
+      expect(frame.roll1).toEqual(null);
+      expect(frame.roll2).toEqual(null);
+      expect(frame.frameScore).toEqual(null);
+      expect(frame.strike).toEqual(false);
+      expect(frame.spare).toEqual(false);
+      expect(frame.frameFinished).toEqual(false);
+    })
+  })
+
+  describe('bowl', () => {
+    describe('first frame', () => {
+      let frame;
+
+      beforeEach(() => {
+        frame = new Frame;
+        spyOn(window, 'prompt').and.returnValue(1);
+        frame.bowl();
+        frame.bowl();
+      })
+  
+      it('requests a score for first roll if first roll is null', () => {
+        expect(frame.roll1).toEqual(1);
+      })
+  
+      it('requests a score for second roll if first roll is not null', () => {
+        expect(frame.roll2).toEqual(1);
+      })
```

Comment:
Do these methods (`roll1`, `roll2`) exist for the interface? Or is there another reason they are surfaced? It feels like the inside of the Frame is not super encapsulated if all of it is made to be accessed.